### PR TITLE
feat(cli): implement search command

### PR DESCRIPTION
**Pull request by beeman**
Quick implementation of a search command to allow searching from the CLI.

Open for any feedback :)

### CODE REVIEW

File: cli/lib/actions.ts [hunk 1]
```diff
@@ -122,6 +123,34 @@ export const executeCommand = async (script: string) => {
...
+        }
+      });
+
+      table.push(
+        ...res.map(item => ([item.name, item.requestedVersion, item.tags.join(' '), item.owner]))
+      );
+
+      return console.log(table.toString());
+    }
+    return warn(`You searched for ${query}, but that's something I can't find...`)
```
Comment (hweeks) [inline cli/lib/actions.ts:147]: thanks for keeping my message tone :)

File: cli/lib/api/search.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ interface SearchResult {
   tags: string[]
 }
 
-export const search = async (searchTerm : string, searchVersion: string, baseUrl: string) : Promise<string|undefined> => {
+export const search = async (searchTerm : string, searchVersion: string, baseUrl: string) : Promise<string|undefined|any> => {
```
Comment (hweeks) [inline cli/lib/api/search.ts:10]: I'm generally hesistant on any types for stuff, did a type get all dumb on you?

Comment (hweeks) [reply cli/lib/api/search.ts:10]: I see the issue here, If we update the `SearchResult` interface to include `owner: string` you can remove both `any` and clear your errors.

Comment (beeman) [reply cli/lib/api/search.ts]: I updated the SearchResult interface to now include owner (and the `about` field was missing too), that removed the typing issue.

**Comment by beeman** [comment]
> Looks good to me, do we want to support allowing a decision to be amde from the list of results and then downloaded automatically?

I don't think I follow what you mean with this?

**Comment by hweeks** [comment]
Fair point, I will try and explain a bit better.

I was thinking this might be useful:

  1. you search for packages
  1. in the results list you can select a package from your search
  1. once that version is highlighted you can hit enter to download

I'm happy to merge this is as is and work on that enhancement later. This is killer like it is so I'd like to bring it in now. I'll create an associated issue ticket and then we can add that feature later :shipit:

File: cli/package.json [hunk 3]
```diff
@@ -34,8 +34,10 @@
     "test:debug": "node --inspect-brk $(npm bin)/jest --watchAll --no-cache --runInBand"
   },
   "dependencies": {
+    "@types/cli-table": "^0.3.0",
```
Comment (hweeks) [inline cli/package.json:37]: Can we move this to a dev dependency? I'm building the code before publishing, so the types packages are only needed during development.

Comment (beeman) [reply cli/package.json]: Done!

**Comment by beeman** [comment]
> Fair point, I will try and explain a bit better.
> 
> I was thinking this might be useful:
> 
> 1. you search for packages
> 2. in the results list you can select a package from your search
> 3. once that version is highlighted you can hit enter to download

Yeah, that sounds like an awesome addition!

> I'm happy to merge this is as is and work on that enhancement later. This is killer like it is so I'd like to bring it in now. I'll create an associated issue ticket and then we can add that feature later :shipit:

Awesome, thanks for merging and publishing this all so quickly!

**Pull request merged**